Dockerfile

```diff
@@ -20,20 +26,35 @@ RUN yum update -y \
         cyrus-sasl-devel \
         libxml2-devel \
         libxslt-devel \
-        nodejs \
         zlib-devel \
-        mysql-community-libs-compat-5.6.14-3.el6.x86_64 \
+        libffi-devel \
+        openssl-devel \
+        python-devel \
+        # Git, because we're using git-checkout dependencies
+        git \
+        # Nodejs for less, stylus, uglifyjs and others
+        nodejs \
+        # Dependencies for mysql-python
+        mysql-community-devel \
+        mysql-community-client \
+        mysql-community-libs \
     && yum clean all
 
-# The version in the above image is ancient, and does not support the
-# --no-binary flag used in our requirements files.
-# We also need to install wheels.
-RUN pip install -U pip wheel
+# Unfortunately it seems pip isn't available in the centos repository.
+RUN curl https://raw.githubusercontent.com/pypa/pip/master/contrib/get-pip.py > get-pip.py \
```

Comment:
What do ops do to get pip? 

Looks like it's in EPEL (Extra Packages for Enterprise Linux) see https://fedoraproject.org/wiki/EPEL
You can install that repo with the `epel-release` package and then pip  is installable as python-pip.

Then you should be able to install `python-pip` from yum.